invenio_records_resources/services/records/components.py

```diff
@@ -133,3 +135,20 @@ class RelationsComponent(ServiceComponent):
     def read(self, identity, record=None):
         """Read record handler."""
         record.relations.dereference()
+
+
+class BackRelationsComponent(ServiceComponent):
+    """Back Relations service component."""
+
+    def update(self, identity, data=None, record=None, uow=None, **kwargs):
+        """Register a task for the update propagation."""
+        # FIXME: we need a better way... service.name would be ok
+        # or registry.service_id(self.service)
+        record_type = str(record.__class__.__name__).lower() + "s"
```

Comment:
Look into this

Reply:
👍 Or simply an attribute on the record class itself.

Reply:
This variable value needs to match the "service name" given upon registration, currently done at views.py level in invenio-rdm-records. This leads to:
- `record_type` might not be the best variable name
- we have the need to set it somewhere, and it will be used for "registering change notification handlers", "services" and "indexers"

WDYT?

Reply:
I like the fact that the service has become "the one place" that brings a lot of things together (record class, indexer, permissions, etc.), so IMO the service ID (i.e. `names`, `users`, etc.) feels like a good lookup key. Also we're going with components and service methods to implement these notifications/update-propagation, so we're anyways on the service level already.

I think adding `.name` or `.id` to the service, would be the best, and would also shorten the service registration call to just `current_service_registry.register(users_service)`.

Reply:
Done, there was already a `self.config.service_id` that was allowed in the service registry. I made it consistent as part of the base config (and set it for files and records). Means we will have to add this across services but is not heavy.